tools/topology/topology2/include/pipelines/cavs/chain-dma-playback.conf

```diff
@@ -0,0 +1,128 @@
+#
+# CAVS chained DMA playback pipeline
+#
+# A chain DMA pipeline. All attributes defined herein are namespaced
+# by alsatplg to "Object.Pipeline.passthrough-playback.N.attribute_name"
+#
+# Usage: passthrough-playback pipeline object can be instantiated as:
+#
+# Object.Pipeline.chain-dma-playback."N" {}
+#
+# Where N is the unique pipeline ID within the same alsaconf node.
+#
+
+<include/components/copier.conf>
+<include/components/pipeline.conf>
+<include/common/audio_format.conf>
+
+Class.Pipeline."chain-dma-playback" {
+
+	DefineAttribute."index" {}
+
+	<include/pipelines/pipeline-common.conf>
+
+	attributes {
+		!constructor [
+			"index"
+		]
+
+		!immutable [
+			"direction"
+		]
+
+		#
+		# passthrough-playback objects instantiated within the same alsaconf node must have
+		# unique pipeline_id attribute
+		#
+		unique	"instance"
+	}
+
+	Object.Widget {
+		copier."1" {
+			copier_type	"host"
+			node_type $HDA_HOST_OUTPUT_CLASS
+			type	"aif_in"
+			num_audio_formats 3
+			# 16-bit input format 48KHz 2ch. Output sample format is always 32-bit for playback
+			Object.Base.audio_format.1 {
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+			# 24-bit 48KHz 2ch
+			Object.Base.audio_format.2 {
+				in_bit_depth		32
+				in_valid_bit_depth	24
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$obs * 2]"
+			}
+			# 32-bit 48KHz 2ch
+			Object.Base.audio_format.3 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+		}
+		copier."2" {
+			type dai_in
+			dai_type "HDA"
+			copier_type "HDA"
+			node_type $HDA_LINK_OUTPUT_CLASS
+			num_audio_formats 3
+			# 32-bit 48KHz 2ch
+			Object.Base.audio_format.3 {
+				in_bit_depth            32
+				in_valid_bit_depth      24
+				out_bit_depth           32
+				out_valid_bit_depth     24
+				dma_buffer_size "$[$obs * 2]"
+			}
+
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$obs * 2]"
+			}
+			# 32-bit 48KHz 4ch
+			Object.Base.audio_format.2 {
+				in_channels		4
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_channels		4
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$obs * 2]"
+				in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				in_ch_map	$CHANNEL_MAP_3_POINT_1
+				out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				out_ch_map	$CHANNEL_MAP_3_POINT_1
```

Comment:
what is all this? HDMI use of 4ch is rather rare.

Reply:
same comment as above.